### PR TITLE
[UR][CTS] Add tests for urPlatformGetNaitiveHandle and urPlatformCreateWithNativeHandle

### LINKED ISSUE #673: Add CTS tests for `urPlatformCreateWithNativeHandle`
We should add conformance tests for `urPlatformCreateWithNativeHandle`.

Required Coverage:

-  At least one successful (i.e. called with correct arguments and assert on expected result, not applicable to `CreateWithNativeHandle` entry points)
-  If this a query entry-point or one which accepts flags etc : we should tests every possible enumeration (where possible).
-  A test case for each returned error code (exclude disaster situations i.e. OUT_OF_MEMORY etc)

### PR DESCRIPTION
**Pull request by callumfare**
Resolves #673 and #674

### CODE REVIEW

File: test/conformance/platform/urPlatformCreateWithNativeHandle.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "fixtures.h"
+
+using urPlatformCreateWithNativeHandleTest = uur::platform::urPlatformTest;
+
+TEST_F(urPlatformCreateWithNativeHandleTest, InvalidNullPlatformPointer) {
+    ur_native_handle_t native_handle = nullptr;
+    ASSERT_SUCCESS(urPlatformGetNativeHandle(platform, &native_handle));
```
Comment (veselypeta) [inline test/conformance/platform/urPlatformCreateWithNativeHandle.cpp:12]: Don't change this as part of this PR - but I think we could have some logic here to skip this test if the entry-point return `UR_RESULT_ERROR_UNSUPPORTED_FEATURE` lots of adapters do not support these entry-points.

Comment (veselypeta) [reply test/conformance/platform/urPlatformCreateWithNativeHandle.cpp:12]: We actually do something similar is `urDeviceGetInfo` tests

**Review by veselypeta — APPROVED** [review]

File: test/conformance/platform/urPlatformCreateWithNativeHandle.cpp (hunk 1 above)
Comment (callumfare) [reply test/conformance/platform/urPlatformCreateWithNativeHandle.cpp:12]: Yeah that makes sense, I could go over all the GetNative/CreateWithNative entry points in a separate PR to make that consistent

File: test/conformance/platform/urPlatformCreateWithNativeHandle.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "fixtures.h"
+
+using urPlatformCreateWithNativeHandleTest = uur::platform::urPlatformTest;
+
+TEST_F(urPlatformCreateWithNativeHandleTest, InvalidNullPlatformPointer) {
```
Comment (aarongreig) [inline test/conformance/platform/urPlatformCreateWithNativeHandle.cpp:10]: nitpick but `NullPointerPlatform` is more consistent with other tests like this

**Pull request merged**